web_responsive/static/src/css/web_responsive.scss

```diff
@@ -131,16 +131,46 @@ $chatter_zone_width: 35%;
         }
     }
 }
+.o_main_navbar {
+    color: color-yiq($o-brand-odoo);
 
+    > ul > li > a,
+    > ul > li > label {
+        color: color-yiq($o-brand-odoo);
+    }
+    .dropdown-menu.show {
+        max-height: calc(100vh - #{$o-navbar-height});
+    }
+}
 // Iconized full screen apps menu
 .o_menu_apps {
     .search-input:focus {
         border-color: $o-brand-primary;
     }
 
+    a.full {
+        width: $o-navbar-height;
+        text-align: center;
+    }
+
     .dropdown-menu.show {
+        visibility: visible;
+        opacity: 1;
+    }
+
+    .dropdown-menu {
         @include full-screen-dropdown();
 
+        visibility: hidden;
```

Comment:
Again, no need to deal with this:

```suggestion
```

Reply:
It is necessary to keep the visibility property because when opacity is zero the menu is invisible but it is still there. Visibility is more similar to applying a display: none;

Reply:
As it is necessary to leave the visibility property, it is also necessary to "animate" the visibility property since if it changes without a delay, the menu disappears instantly.

Reply:
You can read this stackoverflow answer that explains this very well
https://stackoverflow.com/questions/27900053/css-transition-with-visibility-not-working/37905071#37905071

Reply:
That is a good theory, but in practice, you can see that it's not needed:

![Peek 10-08-2020 08-59](https://user-images.githubusercontent.com/973709/89762208-e0151500-dae7-11ea-808e-056ad7c19ef3.gif)

The difference between `visibility: invisible` and `display: none` is that an invisible block still occupies its space, while an undisplayed block doesn't.

So, this property is unneeded because Bootstrap itself is correctly managing the `display` property, and making an undisplayed block invisible is useless because it's not going to be displayed anyway. Since `display` is not animated, it changes instantly, before `opacity` starts being animated. And that's all you really need here, so `visibility` is not needed.

Reply:
Yes, if you do this the menu effect works, and visibility is not needed, but since the display:flex property is specified, bootstraps is not managing it.

If i only manage the opacity and no the visibility (or display) property, the fade effect works, but the menu is not disappearing. You can't use the apps ui because the invisible menu is on top of it. And animation is not possible if I use display: flex / none because the change is instantaneous and the animation does not have the effect it should.

Reply:
Oh I see now! :+1: then, thanks!